ui/redux/reducers/memberships.js

```diff
@@ -0,0 +1,197 @@
+// @flow
+import * as ACTIONS from 'constants/action_types';
+
+const reducers = {};
+
+type MembershipsState = {
+  membershipMineByKey: ?MembershipMineDataByKey,
+  membershipListById: { [channelId: string]: MembershipTiers },
+  channelMembershipsByCreatorId: ChannelMembershipsByCreatorId,
+  fetchingIdsByCreatorId: { [creatorId: string]: ClaimIds },
+  pendingBuyIds: ClaimIds,
+  pendingCancelIds: ClaimIds,
+  myMembershipTiers: ?MembershipTiers,
+  pendingDeleteIds: Array<string>,
+  protectedContentClaimsByCreatorId: { [channelId: string]: any },
+  mySupportersList: ?SupportersList,
+  membershipPerks: Array<MembershipPerk>,
+  listingAllMyTiers: ?boolean,
+};
+
+const defaultState: MembershipsState = {
+  membershipMineByKey: undefined,
+  membershipListById: {},
+  channelMembershipsByCreatorId: {},
+  fetchingIdsByCreatorId: {},
+  pendingBuyIds: [],
+  pendingCancelIds: [],
+  myMembershipTiers: undefined,
+  pendingDeleteIds: [],
+  protectedContentClaimsByCreatorId: {},
+  mySupportersList: undefined,
+  membershipPerks: [],
+  listingAllMyTiers: undefined,
+};
+
+reducers[ACTIONS.CHANNEL_MEMBERSHIP_CHECK_STARTED] = (state, action) => {
+  const { channel, ids } = action.data;
+  const { fetchingIdsByCreatorId: currentFetching } = state;
+
+  return { ...state, fetchingIdsByCreatorId: { ...currentFetching, [channel]: ids } };
+};
+reducers[ACTIONS.CHANNEL_MEMBERSHIP_CHECK_COMPLETED] = (state, action) => {
+  const { channelId, membershipsById } = action.data;
+
+  const currentFetched = Object.assign({}, state.channelMembershipsByCreatorId);
+  const newFetched = currentFetched[channelId] ? { ...currentFetched[channelId], ...membershipsById } : membershipsById;
+  delete state.fetchingIdsByCreatorId[channelId];
+
```

Comment:
:small_red_triangle: Cannot alter `state` directly. 

Must be altered immutably, otherwise the GUI might not detect/respond to changes correctly.